src/test/java/org/sonarlint/intellij/analysis/LocalFileExclusionsTest.java

```diff
@@ -52,38 +54,40 @@ public void prepare() {
   }
 
   @Test
-  public void should_not_analyze_automatically_if_module_is_null() {
+  public void test_should_not_analyze_automatically_if_module_is_null() {
     LocalFileExclusions.Result result = exclusions.canAnalyze(testFile, null);
     assertThat(result.isExcluded()).isTrue();
   }
 
   @Test
-  public void should_analyze_file() {
+  public void test_should_analyze_file() {
     assertThat(exclusions.canAnalyze(testFile, getModule()).isExcluded()).isFalse();
   }
 
   @Test
-  public void should_not_analyze_if_file_is_binary() {
+  public void test_should_not_analyze_if_file_is_binary() {
     when(type.isBinary()).thenReturn(true);
     assertThat(exclusions.canAnalyze(testFile, getModule()).isExcluded()).isTrue();
   }
 
   @Test
-  public void should_not_analyze_if_module_is_null() {
+  public void test_should_not_analyze_if_module_is_null() {
     assertThat(exclusions.canAnalyze(testFile, null).isExcluded()).isTrue();
   }
 
-//// TODO maybe use heavy test
-//  @Test
-//  public void should_not_analyze_if_project_is_disposed() {
-//    getProject().dispose();
-//    ApplicationManager.getApplication().runWriteAction(() -> getProject().dispose());
-//    ProjectUtil.closeAndDispose(getProject());
-//    assertThat(exclusions.canAnalyze(testFile, getModule()).isExcluded()).isTrue();
-//  }
+  @Test
+  public void test_should_not_analyze_if_project_is_disposed() throws IOException, JDOMException {
+    String projectFilePath = myProject.getProjectFilePath();
+    PlatformTestUtil.forceCloseProjectWithoutSaving(myProject);
+
+    assertThat(exclusions.canAnalyze(testFile, getModule()).isExcluded()).isTrue();
+
+    // reopen to avoid error
+    myProject = ProjectManager.getInstance().loadAndOpenProject(projectFilePath);
```

Comment:
Shouldn't it be in a finally block?

Reply:
Or maybe in a tear down method ? If the project is closed, we reopen it for the next test

Reply:
Yep, even better